### ISSUE TITLE
Can I get the position of a character to align an annotation?

### ISSUE THREAD

**Issue by greaber** [facebook/draft-js#560]
I have an editor where the user can highlight a region of text and create an annotation that is aligned with the highlight.  It is implemented using CodeMirror, which exposes methods `charCoords` and `cursorCoords` that give the coordinates of a bounding box around, respectively, the character at a given position in the editor, and an imaginary cursor at a given position in the editor. When the editor contents change, CodeMirror emits an event, and I recheck all the character positions I am tracking and adjust the positions of the annotations as necessary.  I am investigating switching to draft.js, but I couldn't find any API to implement this functionality in the docs.  Do I need to fish spans corresponding to the highlights out of the DOM and call `getBoundingClientRect` on them?

**Comment by zhongkai** [comment]
Same puzzle to me. I only find that there is a method to identify block level position, not character.

**Comment by roundrobin** [comment]
+ @flarnie Could you help us with this question?

**Comment by kedromelon** [comment]
> Do I need to fish spans corresponding to the highlights out of the DOM and call getBoundingClientRect on them?

yes -- (if i understand the question correctly) as far as i know this is the best solution here. we do something along these lines in our application using draftjs.
